Support TwitchEmote's v4 API, BTTV v3 API, upgrade deps

mkody opened the pull request:
## NOTE

This PR needs to be updated to drop TwitchEmote altogether as its API doesn't work anymore.  
[My fork](https://github.com/mkody/twitch-emoticons) now uses the Twitch API directly, instead.

Since this repo appears abandoned, I've published a package from my `fork` branch: [@mkody/twitch-emoticons](https://www.npmjs.com/package/@mkody/twitch-emoticons)

---

The TwitchEmote v3 API got shut down. Same for BTTV v2.

I pretty much reused bits from the BTTV part for this.  
Oh and also fixed the quotes in the html template.

**BREAKING CHANGES:** `fetchTwitchEmotes` takes the ID of the channel as a parameter instead of a string or array of strings with the channel name.  
**BREAKING CHANGES 2:** `fetchBTTVEmotes` takes the ID of the channel as a parameter instead of a string of the channel name.  
You should also use a Twitch ID for FFZ, but support for both is still available _for now_.

Some dependencies and dev-dependencies were updated too.

mkody committed:
`null` in fetchTwitchEmotes crashes

Also it tries to test against `channel` instead of `channel_name`

mkody committed:
Fix ident

mkody committed:
Support TwitchEmotes's v4 API

Also fixed the html template.

mkody committed:
Fix tests

ngajhede committed:
Update package.json

ngajhede committed:
Update package.json

ngajhede committed:
Chanegd to request-promise

ngajhede committed:
Chanegd to request-promise

ngajhede committed:
Changed to request-promise

ngajhede committed:
Changed to request-promise

ngajhede committed:
Update EmoteFetcher.js

ngajhede committed:
Update EmoteFetcher.js

ngajhede committed:
Update EmoteFetcher.js

ngajhede committed:
Update EmoteFetcher.js

mkody committed:
Upgrade devdeps and fix lint

mkody committed:
Build new docs

mkody committed:
Merge branch 'ngajhede'

mkody committed:
Missing request dep

mkody committed:
update typings

mkody committed:
Link to my repo

Nisgrak commented:
I think the lib is abandoned. Maybe you can upload the lib as another package?

mkody commented:
I could look into that but in the meantime you can install from git which works well.
`yarn add https://github.com/mkody/twitch-emoticons.git`

AhiVT commented:
@mkody May I adopt your fork as maintainer and upload to NPM? It would be very beneficial to other developers working with Twitch emotes on their sites.

mkody committed:
Publish

mkody committed:
Oops, left a typo in the README

mkody commented:
I went and did it: https://www.npmjs.com/package/@mkody/twitch-emoticons

dependabot[bot] committed:
Bump lodash from 4.17.15 to 4.17.20

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.20.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.20)

Signed-off-by: dependabot[bot] <support@github.com>

ssut committed:
fix(typings): module name

mkody committed:
Should support BTTV v3 API; Support ID for FFZ

The latter now makes it easier to just use the Twitch ID for everything.

mkody committed:
Use Github Actions instead of travis; Edit README

mkody committed:
Bump version; Upgrade Deps; Gen docs

mkody committed:
Add important note in README

mkody committed:
I copy/pasted that line a bit too fast

mkody committed:
Fix require to use in README/docs

mkody committed:
Update  package-lock.json

I really don't use npm by itself so... oops.

mkody committed:
Remove package-lock since we're working with yarn

mkody committed:
Replace request (Close #9); Fallback for Global

If twitchemotes.com can't return global emotes (down or broken),
use a static json I'm hosting as a fallback.

mkody committed:
Update docs and bump version

mkody committed:
Add dependabot

mkody committed:
Fix missing input?

dependabot[bot] committed:
Bump eslint from 7.18.0 to 7.25.0 (#10)

Bumps [eslint](https://github.com/eslint/eslint) from 7.18.0 to 7.25.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.18.0...v7.25.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump eslint from 7.25.0 to 7.26.0 (#11)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mkody committed:
Remove normal dependabot or it will be too noisy

Killusions committed:
fix: twitch emote fetching (#12)

mkody committed:
Change fallback request to `.body` too

mkody committed:
Bump version; Upgrade deps; Regen docs

Killusions committed:
feat: use twitch api

mkody committed:
Merge branch 'fork' into feat/use-twitch-api

mkody committed:
Update tests; Regen docs

mkody committed:
Regen HTML docs

mkody committed:
Minor update to tests; Load secrets to GH Actions

mkody committed:
Always test fetchTwitchEmotes without API keys

mkody committed:
Fix typings; Update readme and contributors

I hope this is all we need to fix typings actually.

mkody committed:
Upgrade deps; Split BTTV and FFZ tests

dependabot[bot] committed:
Bump got from 11.8.3 to 11.8.5 (#16)

Bumps [got](https://github.com/sindresorhus/got) from 11.8.3 to 11.8.5.
- [Release notes](https://github.com/sindresorhus/got/releases)
- [Commits](https://github.com/sindresorhus/got/compare/v11.8.3...v11.8.5)

---
updated-dependencies:
- dependency-name: got
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mkody committed:
Use new Twitch CDN template

Close Support Twitch animated icons mkody/twitch-emoticons#17

mkody committed:
Fix Twitch test

mkody committed:
Upgrade deps

mkody committed:
Bump to 2.5.0

mkody committed:
FFZ is now ID-only; Add 7TV support

And update JSDoc types, typings

7TV issue: mkody/twitch-emoticons#15

mkody committed:
Fix typo in typings

mkody committed:
Fix class name: SevenTV => SevenTVEmote

Also bump deps

mkody committed:
Add support for 7tv with the v3 API

Related issue: #15

mkody committed:
Add more examples

mkody committed:
Bump to 2.6.0

And update typings

mkody committed:
Update docs with correct version

mkody committed:
Create codeql.yml

mkody committed:
Create eslint.yml

mkody committed:
CodeQL: Ignore test and docs paths
